### PR TITLE
Add getClaimableBalanceId() to determine claimable balance ids

### PR DESCRIPTION
**Pull request by tamirms**
Fixes https://github.com/stellar/java-stellar-sdk/issues/299

### CODE REVIEW

File: src/main/java/org/stellar/sdk/Transaction.java [hunk 1]
```diff
@@ -92,6 +94,37 @@ public long getFee() {
     return mOperations;
   }
 
+  /**
+   * Returns returns the claimable balance ID for the CreateClaimableBalanceOperation at the given index within the transaction.
```
Comment (Shaptic) [inline src/main/java/org/stellar/sdk/Transaction.java:98]: nit: "Returns returns"

File: src/main/java/org/stellar/sdk/Transaction.java [hunk 2]
```diff
@@ -92,6 +94,37 @@ public long getFee() {
...
+   */
+  public String getClaimableBalanceId(int index) throws IOException {
+    if (index >= mOperations.length) {
```
Comment (Shaptic) [inline src/main/java/org/stellar/sdk/Transaction.java:101]: nit: do we need to check < 0 too? or `uint`ify the index

File: src/test/java/org/stellar/sdk/ClaimableBalanceIdTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        )).build();
+    Transaction transaction = new Transaction.Builder(AccountConverter.enableMuxed(), new Account(sourceAccount, 123l), Network.TESTNET)
+        .addOperation(op0)
+        .addOperation(new BumpSequenceOperation.Builder(2l).build())
+        .addOperation(op0)
+        .setTimeout(Transaction.Builder.TIMEOUT_INFINITE)
+        .setBaseFee(Transaction.MIN_BASE_FEE)
+        .build();
+
+    String expectedIdIndex0 = "0000000095001252AB3B4D16ADBFA5364CE526DFCDA03CB2258B827EDBB2E0450087BE51";
```
Comment (Shaptic) [inline src/test/java/org/stellar/sdk/ClaimableBalanceIdTest.java:37]: Why are they uppercase? I don't think `GET /claimable_balance/:id` is case-agnostic, so we'd want to make sure they're compatible.

File: src/main/java/org/stellar/sdk/Transaction.java (hunk 1 above)
Comment (Shaptic) [reply src/main/java/org/stellar/sdk/Transaction.java:98]: Can we also add a cross-reference to the docstring here? Either to the relevant [Stellar Core code](https://github.com/stellar/stellar-core/blob/9f3cc04e6ec02c38974c42545a86cdc79809252b/src/test/TestAccount.cpp#L285) or to the [txnbuild implementation](https://github.com/stellar/go/blob/master/txnbuild/transaction.go#L356). It'll be super helpful if we need to return to this code.

File: src/test/java/org/stellar/sdk/ClaimableBalanceIdTest.java (hunk 3 above)
Comment (Shaptic) [reply src/test/java/org/stellar/sdk/ClaimableBalanceIdTest.java:37]: Either in the encoding or in the request builder, not sure what's more appropriate

File: src/main/java/org/stellar/sdk/Transaction.java [hunk 4]
```diff
@@ -92,6 +94,37 @@ public long getFee() {
...
+    body.setSeqNum(new SequenceNumber(new Int64(getSequenceNumber())));
+    body.setSourceAccount(AccountConverter.disableMuxed().encode(getSourceAccount()));
+    id.setId(body);
+
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    id.encode(new XdrDataOutputStream(outputStream));
+
+    ClaimableBalanceID result = new ClaimableBalanceID();
+    result.setDiscriminant(ClaimableBalanceIDType.CLAIMABLE_BALANCE_ID_TYPE_V0);
+    result.setV0(new Hash(Util.hash(outputStream.toByteArray())));
```
Comment (Shaptic) [inline src/main/java/org/stellar/sdk/Transaction.java:121]: This is probably obvious with context & usage but there's no risk of the hash somehow... not being sha2 at some point?

**Review by Shaptic — APPROVED** [review]
A few minor notes, but LGTM otherwise! :+1:

File: src/main/java/org/stellar/sdk/Transaction.java (hunk 4 above)
Comment (tamirms) [reply src/main/java/org/stellar/sdk/Transaction.java:121]: unless we decide to to swap out sha2 for a different algorithm for transaction hashing it should remain as sha2

**Pull request merged**